js/main.js

```diff
@@ -0,0 +1,142 @@
+
+class Card {
+    constructor(effect, mode) {
+        this.effect = effect;
+        this.mode = mode;
+        this.faceUp = false;
+    }
+    flipUp() {
+        this.faceUp = true;
+    }
+    flipDown() {
+        this.faceUp = false;
+    }
+}
+
+const table = [];
+let isComplete;
+let turn;
+let selectedCards = [];
+
+const cardSlots = document.querySelectorAll('figure');
+const selectTable = document.querySelector('main');
+const selectComplete = document.getElementById('complete')
+
+selectTable.addEventListener('click', cardFlip);
+
+function startGame() {
+    const effects = ['bold', 'italics', 'bullet', 'code', 'h1', 'h2', 'link', 'number'];
+    const modes = ['markdown', 'format'];
+    isComplete = false;
+    turn = 1;
+
+    for (const effect of effects) {
+        for (const mode of modes) {
+            table.push(new Card(effect, mode));
+        }
+    }
+
+    table.sort(() => Math.random() - 0.5)
+
+    tableSetup();
+}
+
+function tableSetup() {
+    for (let i = 0; i < table.length; i++) {
+        document.getElementById(`card${i}`).classList.add(`${table[i].effect}${table[i].mode}`, "back");
+    }
+}
+
+function cardFlip(event) {
+
+    const idx = parseInt(event.target.id.replace('card', ''));
+
+    if (table[idx].faceUp === false) {
+        table[idx].flipUp();
+        displayTable();
+        selectedCards.push(idx);
+
+        if (turn === 1) {
+            turn *= -1;
+        } else if (turn === -1) {
+            compareCards();
+            testCompletion();
+            turn *= -1;
+        }
+
+    }
+}
+
+
+function displayTable() {
+    for (let i = 0; i < table.length; i++) {
+        const selectCard = document.getElementById(`card${i}`);
+
+        if (table[i].faceUp === false) {
+            selectCard.classList.add("back");
+        } else {
+            selectCard.classList.remove("back");
+        }
```

Comment:
Tip: `card.faceUp` is a boolean value. You can remove the equality check from the condition and evaluate `table[i].faceUp` directly in the if statement. This is what it would look like without a comparison:

```js
if (table[i].faceUp) {
  selectCard.classList.remove("back");
} else {
  selectCard.classList.add("back");
}
```

Note that I changed the order of the branches in the if statement. This is because the main branch is executed when `table[i].faceUp` is true, and your original condition was checking for a false value.

If you wanted to keep checking for a negative value: `false`, you can also use the `!` (NOT) operator, which reverses the boolean value. This would mean your code could be simplified to:  

```js
if (!table[i].faceUp) {
  selectCard.classList.add("back");
} else {
  selectCard.classList.remove("back");
}
```

Generally, I find positive conditions easier to read, but it always depends :)